pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -319,422 +320,469 @@ async def run(self):
         remote_url = remote_url if remote_url[-1] == "/" else f"{remote_url}/"
         optimize_sync = self.optimize
 
+        task = asyncio.ensure_future(self._fetch_remote_urls(remote_url))
+        await asyncio.wait({task})
+
         progress_data = dict(message='Downloading Metadata Files', code='downloading.metadata')
         with ProgressReport(**progress_data) as metadata_pb:
-            downloader = self.remote.get_downloader(
-                url=urljoin(remote_url, 'repodata/repomd.xml')
-            )
-            # TODO: decide how to distinguish between a mirror list and a normal repo
-            result = await downloader.run()
-            metadata_pb.increment()
-
-            repomd_path = result.path
-            repomd = cr.Repomd(repomd_path)
-
-            # Caution: we are not storing when the remote was last updated, so the order of this
-            # logic must remain in this order where we first check the version number as other
-            # changes than sync could have taken place such that the date or repo version will be
-            # different from last sync
-            if (
```

Comment:
There's going to be some nasty conflicts between this PR and Lubos' PR https://github.com/pulp/pulp_rpm/pull/1707

Reply:
Do not worry.